### PR TITLE
Feat/add player to blacklist

### PR DESCRIPTION
**Pull request by lucasmonstro**
## Status
READY

## Migrations
NO

## Description

Add feature to add player to blacklist

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Steps to Test or Reproduce

Follow guidelines and execute addPlayerToBlacklist mutation on graphql endpoint

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/api/src/app/auth/guards/gql-auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ExecutionContext, Injectable } from '@nestjs/common';
+import { GqlExecutionContext } from '@nestjs/graphql';
+import { AuthGuard } from '@nestjs/passport';
+@Injectable()
+export class GqlAuthGuard extends AuthGuard('jwt') {
+  getRequest(context: ExecutionContext) {
```
Comment (lucasmonstro) [inline apps/api/src/app/auth/guards/gql-auth.guard.ts:6]: add return type or todo

**Review by felinto-dev — CHANGES_REQUESTED** [review]

File: apps/api/src/app/blacklist/resolvers/add-player-to-blacklist/add-player-to-blacklist.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export class AddPlayerToBlacklistResolver {
+  constructor(
+    private addPlayerToBlacklistService: AddPlayerToBlacklistService
+  ) {}
+  @Mutation(() => Boolean)
+  async addPlayerToBlacklist(
+    @CurrentUser('guid') ownerFaceitId: string,
+    @Args('input') input: AddPlayerToBlacklistInput
+  ): Promise<true> {
+    return await this.addPlayerToBlacklistService.add(ownerFaceitId, input);
```
[the remark this answers is not in the document]
Comment (lucasmonstro) [reply apps/api/src/app/blacklist/resolvers/add-player-to-blacklist/add-player-to-blacklist.resolver.ts:15]: done

File: apps/api/src/app/blacklist/resolvers/add-player-to-blacklist/add-player-to-blacklist.resolver.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      AddPlayerToBlacklistResolver
+    );
+    service = module.get<AddPlayerToBlacklistService>(
+      AddPlayerToBlacklistService
+    );
+  });
+  it('should add player to blacklist', async () => {
+    const ownerFaceitId = faker.random.uuid();
+    const addPlayerToBlacklistInput = {
+      faceitId: faker.random.uuid(),
```
[the remark this answers is not in the document]
Comment (lucasmonstro) [reply apps/api/src/app/blacklist/resolvers/add-player-to-blacklist/add-player-to-blacklist.resolver.spec.ts:24]: done

File: apps/api/src/app/blacklist/services/add-player-to-blacklist/add-player-to-blacklist.service.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      throw new CannotAddOwnerToYourOwnBlacklistException();
+    }
+    await this.userModel.updateOne(
+      {
+        faceitId: ownerFaceitId,
+        'blacklistedPlayers.faceitId': { $ne: input.faceitId },
+      },
+      { $push: { blacklistedPlayers: input } },
+    );
+    return true;
```
[the remark this answers is not in the document]
Comment (lucasmonstro) [reply apps/api/src/app/blacklist/services/add-player-to-blacklist/add-player-to-blacklist.service.ts:28]: done

**Review by felinto-dev — CHANGES_REQUESTED** [review]

**Review by felinto-dev — APPROVED** [review]

**Pull request merged**